java/mykola-melnyk/src/com/zoolatech/lecture6/tasks/_3/UserRepository.java

```diff
@@ -0,0 +1,28 @@
+package com.zoolatech.lecture6.tasks._3;
+
+import java.util.Optional;
+
+public class UserRepository {
+    private UserCache myUserCache;
+    private UserTable myUserTable;
+
+    public UserRepository(UserCache cache, UserTable table) {
+        myUserCache = cache;
+        myUserTable = table;
+    }
+
+    public String findUserEmail(String id) throws UserMissingException{
+        Optional<User> cachedUser = myUserCache.findUser(id);
```

Comment:
try to simplify you code using Optional's methods `or...` and `map`

Reply:
Taras, can you please give me a hint here? I have tried to figure this out myself, but it seems to be too complicated with .map transformation and how to combine these 2 methods... :(